### PR TITLE
feat: provides access to the CSS variable created by CSSCustomProprtyBehaviors

### PR DESCRIPTION
**Pull request by nicholasrice**
# Description
Adds a `var` field to CSSCustomPropertyBehaviors to provide the CSS Custom Property formatted as a CSS variable. This allows the string to be interpolated into stylesheets.

Also integrates the interpolation into stylesheets.

**Question:**
I waffled between `CSSCustomPropertyBehavior.var` and `CSSCustomPropertyBehavior.css`. I want to keep the property concise to benefit from minification processes but wasn't sure which felt more intuitive. If anyone prefers one or the other (or something else) please let me know, I'm not married to `CSSCustomPropertyBehavior.var`.
## Motivation & context
Primary changes are in [packages/web-components/fast-foundation/src/custom-properties/behavior.ts](https://github.com/microsoft/fast-dna/pull/3256/files#diff-fab8b531919d6f848d435c387e8f1981)

## Issue type checklist

- [ ] **Chore**: A change that does not impact distributed packages.
- [ ] **Bug fix**: A change that fixes an issue, link to the issue above.
- [x] **New feature**: A change that adds functionality.

**Is this a breaking change?**
- [ ] This change causes current functionality to break.

**Adding or modifying component(s) in `@microsoft/fast-components` checklist**

- [ ] I have added a new component
- [ ] I have modified an existing component
- [ ] I have updated the [definition file](https://github.com/Microsoft/fast-dna/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/Microsoft/fast-dna/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

## Process & policy checklist

- [ ] I have added tests for my changes.
- [x] I have tested my changes.
- [x] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast-dna/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/en/contributing/standards) for this project.

### CODE REVIEW

File: packages/web-components/fast-foundation/src/custom-properties/behavior.ts [hunk 1]
```diff
@@ -18,8 +18,120 @@ export interface CSSCustomPropertyTarget {
...
+    private host: (
+        source: typeof FASTElement & HTMLElement
+    ) => Partial<CSSCustomPropertyTarget> | null;
+
+    private _propertyName: string;
+    /**
+     * The name of the CSSCustomPropertyBehavior formatted
+     * as a CSS custom property.
+     */
+    public get propertyName() {
```
Comment (EisenbergEffect) [inline packages/web-components/fast-foundation/src/custom-properties/behavior.ts:78]: Any reason not to assign `propertyName `and `var` in the constructor when the name is set?

Comment (nicholasrice) [reply packages/web-components/fast-foundation/src/custom-properties/behavior.ts:78]: Probably not a good one. The idea was to lazily evaluate it but the operation is so cheap that its probably better to just do the work in the constructor, reduce the amount of coded needed to do the work and remove the runtime check.

I'll update.

**Comment by nicholasrice** [comment]
@EisenbergEffect  looks like a 3.244kb reduction in the fast-components.min.js file

**Comment by EisenbergEffect** [comment]
@nicholasrice Nice!

**Pull request merged**